nvi

```diff
@@ -100,32 +116,58 @@ function install_node() {
   mkdir --parents -- "$BIN_DIR"
 
   # Download and extract the tar if it doesn't exist on disk.
-  if [[ ! -d "$INSTALL_DIR/node-v$NODE_VERSION-linux-x64" ]]; then
+  if [[ ! -d "$INSTALL_DIR/$PACKAGE" ]]; then
     # Fail if version doesn't exist.
-    NODE_DIST_LOOKUP=$(wget -qO - https://nodejs.org/dist/index.json | grep --null-data --only-matching --perl-regex --text "\"version\"\s*:\s*\"v$NODE_VERSION\"")
-    [[ -z "$NODE_DIST_LOOKUP" ]] && >&2 echo "Error: Couldn't find Node.js version v$NODE_VERSION. Aborting." && exit 1
+    # --null-data: prevent grep from writing to stdout before wget finishes.
+    wget -qO - https://nodejs.org/dist/index.json |
+    grep \
+      --null-data \
+      --quiet \
+      --perl-regex \
+      "\"version\"\s*:\s*\"v$NODE_VERSION\"" ||
```

Comment:
I would probably prefer this argument to be either on the same line as `--perl-regex` or be indented another level to indicate that it belongs to the parent flag. 

That way it also becomes a bit more obvious that the next line is an alternative command and not just yet another argument for one of the flags.